tools/wptserve/wptserve/config.py

```diff
@@ -28,11 +28,20 @@ def _merge_dict(base_dict, override_dict):
 
 
 class Config(Mapping):
-    """wptserve config
+    """wptserve config. Immutable configuration data that's shared between processes.
 
-    Inherits from Mapping for backwards compatibility with the old dict-based config"""
-    def __init__(self, logger_name, data):
-        self.__dict__["_logger_name"] = logger_name
+    Inherits from Mapping for backwards compatibility with the old dict-based config
+
+    :param logger: - Either the name of the logger to use, or an object that will be
+                     used as the logger. In the case of an oject this must be safe
+                     to share directly between processes.
+    :param data: - Extra configuration data
+    """
+    def __init__(self, logger, data):
+        if isinstance(logger, str):
```

Comment:
It also feels a bit surprising that logger is a str/logger? Can we make the ConfigBuilder always return a (safe) logger object?

Reply:
Is the contract something like the parent process picks a unique logger name, hands that to wptserve, and then parses the stdout to forward the stuff logged with that logger name? And there's the assumption that wptserve is using mozlog with a specific format I guess?